web/static/webcomponents/pages/ApiKeys/ApiKeys.js

```diff
@@ -0,0 +1,71 @@
+import { fromHTML } from "../../utils.js";
+
+const apiKeysTemplate = document.createElement("template");
+apiKeysTemplate.innerHTML = `
+<link rel="stylesheet" type="text/css" href="/static/webcomponents/pages/ApiKeys/ApiKeys.css">
+<div class="api-keys-container">
+    <ul id="api-keys-list"></ul>
+</div>
+`;
+
+class ApiKeys extends HTMLElement {
+  constructor() {
+    super();
+    this.shadow = this.attachShadow({ mode: "open" });
+    this.shadow.appendChild(apiKeysTemplate.content.cloneNode(true));
+    this.apiKeysData = [];
+    this.apiKeyCount = 0;
+  }
+
+  set apiKeys(data) {
+    this.apiKeysData = data;
+    this.apiKeyCount = data.length;
+    this.renderApiKeys();
+  }
+
+  renderApiKeys() {
+    const apiKeysList = this.shadow.getElementById("api-keys-list");
+    apiKeysList.innerHTML = "";
+    this.apiKeysData.forEach((key, index) => {
+      const listItem = document.createElement("li");
+      listItem.innerHTML = `
+        <span>${this.apiKeyCount - index}. ${key.key}</span>
+        <button class="delete-button" data-key="${key.key}">Delete</button>
```

Comment:
for the key of the span, use id instead of key. Keys are sensitive information we don't want them spilling over unnecessarily.